app/assets/javascripts/discourse/app/components/d-modal.js

```diff
@@ -132,13 +132,18 @@ export default Component.extend({
 
     this.set("headerClass", data.headerClass || null);
 
-    if (this.element) {
-      const autofocusInputs = this.element.querySelectorAll(
+    if (this.element && data.autoFocus) {
+      let focusTarget = this.element.querySelector(
         ".modal-body input[autofocus]"
       );
 
-      if (autofocusInputs.length) {
-        afterTransition(() => autofocusInputs[0].focus());
+      if (!focusTarget && !this.site.mobileView) {
```

Comment:
Did this cause issues on mobile?

Reply:
I copied that from the old code I moved. It might not be necessary anymore but we were doing it before for a reason.